### PR TITLE
Bookmark

### PR DESCRIPTION
**Pull request by JatuFaique**
- Added Feat to add/ remove bookmark
- added bookmark page
- updated nav links

### CODE REVIEW

File: src/Components/PostCard.js [hunk 1]
```diff
@@ -41,6 +42,15 @@ function PostCard({ post }) {
     (like) => like._id === userDetail._id
   );
 
+  const isBookmarked = userDetail.bookmarks.some((mark) => mark === post._id);
+
+  const bookmarkHandler = () => {
+    isBookmarked
+      ? dispatch(handleUnBookmark({ postId: post._id, token: token }))
+      : dispatch(handleBookmark({ postId: post._id, token: token }));
+    // dispatch(handleBookmark({ postId: post._id, token: token }));
```
Comment (parulgupta2001) [inline src/Components/PostCard.js:51]: remove commented code

File: src/features/auth/authSlice.js [hunk 2]
```diff
@@ -157,6 +203,28 @@ export const authSlice = createSlice({
       state.errorMessage = action.payload.errors;
       state.status = "Rejected";
     },
+    [handleBookmark.pending]: (state) => {
+      state.status = "loading";
+    },
+    [handleBookmark.fulfilled]: (state, action) => {
+      console.log("ho gayi", action.payload);
```
Comment (parulgupta2001) [inline src/features/auth/authSlice.js:212]: remove console

File: src/pages/Bookmarks.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from "react";
+import notfound from "../assets/not-found.svg";
+import { useDispatch, useSelector } from "react-redux";
+import PostCard from "../Components/PostCard";
```
Comment (parulgupta2001) [inline src/pages/Bookmarks.js:4]: import all these in index file then export from there.

**Review by parulgupta2001 — COMMENTED** [review]
small suggestions are given otherwise everything looks ok

**Review by parulgupta2001 — APPROVED** [review]

File: src/Components/PostCard.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import React, { useState } from "react";
 import { useDispatch, useSelector } from "react-redux";
+import { handleBookmark, handleUnBookmark } from "../features/auth/authSlice";
```
Comment (sonishreyas) [inline src/Components/PostCard.js:3]: can use absolute imports, you can refer this 
https://dev.to/mr_frontend/absolute-imports-in-create-react-app-3ge8

File: src/Components/SideBar.js [hunk 5]
```diff
@@ -30,27 +30,35 @@ function SideBar() {
...
-              <div class="tab px-1 py-0-5">
-                <i class="fa-solid fa-user-tie"></i>
-                <span class="hideMd"> Profile </span>
+              <div className="tab px-1 py-0-5">
+                <NavLink to="/profile">
+                  <i class="fa-solid fa-user-tie"></i>
+                  <span class="hideMd"> Profile </span>
+                </NavLink>
               </div>
             </li>
```
Comment (sonishreyas) [inline src/Components/SideBar.js:63]: This looks like a repetative code you can create an array with the name and endpoint, and loop through it

File: src/backend/controllers/UserController.js [hunk 6]
```diff
@@ -77,6 +77,7 @@ export const editUserHandler = function (schema, request) {
 
 export const getBookmarkPostsHandler = function (schema, request) {
   const user = requiresAuth.call(this, request);
+  console.log("im here", user.bookmarks);
```
Comment (sonishreyas) [inline src/backend/controllers/UserController.js:80]: you can remove this

**Review by sonishreyas — APPROVED** [review]
Code looks good. There are multiple console.log() in many places, you can remove them. Also, had a few suggestion that you can implement.

**Comment by JatuFaique** [comment]
All the suggestions are noted, will try to implement in next PR.
Thanks

**Pull request merged**